Src/LiquidProjections.PollingEventStore/PollingEventStoreAdapter.cs

```diff
@@ -182,13 +182,12 @@ private void StartPreloadingNextPage(long previousCheckpoint, string subscriptio
                     throw new OperationCanceledException();
                 }
 
-                CheckpointRequestTimestamp effectiveLastExistingCheckpointRequest =
-                    Volatile.Read(ref lastSuccessfulPollingRequestWithoutResults);

```

Comment:
If this read is not volatile, there is no guarantee that it will ever read any non-null value in which case it will poll the event store without any delay. Race condition in the old code was not a big problem because it could only happen when the event store had a new transaction just loaded. But at least a memory barrier is required here I think.

The problem on the write side is worse however. The struct is not written atomically. So moving to struct requires a monitor lock.